dipy/workflows/tests/test_reconst_csa_csd.py

```diff
@@ -35,77 +36,82 @@ def reconst_flow_core(flow):
-
-        pam = load_peaks(reconst_flow.last_generated_outputs['out_pam'])
-        npt.assert_allclose(pam.peak_dirs.reshape(peaks_dir_data.shape),
-                            peaks_dir_data)
-        npt.assert_allclose(pam.peak_values, peaks_vals_data)
-        npt.assert_allclose(pam.peak_indices, peaks_idx_data)
-        npt.assert_allclose(pam.shm_coeff, shm_data)
-        npt.assert_allclose(pam.gfa, gfa_data)
-
-        bvals, bvecs = read_bvals_bvecs(bval_path, bvec_path)
-        bvals[0] = 5.
-        bvecs = generate_bvecs(len(bvals))
-
-        tmp_bval_path = pjoin(out_dir, "tmp.bval")
-        tmp_bvec_path = pjoin(out_dir, "tmp.bvec")
-        np.savetxt(tmp_bval_path, bvals)
-        np.savetxt(tmp_bvec_path, bvecs.T)
-        reconst_flow._force_overwrite = True
-        with npt.assert_raises(BaseException):
-            npt.assert_warns(UserWarning, reconst_flow.run, data_path,
-                             tmp_bval_path, tmp_bvec_path, mask_path,
-                             out_dir=out_dir, extract_pam_values=True)
-
-        if flow.get_short_name() == 'csd':
-
-            reconst_flow = flow()
-            reconst_flow._force_overwrite = True
+        for sh_order in [4, 6, 8]:
             reconst_flow.run(data_path, bval_path, bvec_path, mask_path,
-                             out_dir=out_dir, frf=[15, 5, 5])
-            reconst_flow = flow()
+                            sh_order=sh_order,
+                            out_dir=out_dir, extract_pam_values=True)
+
+            gfa_path = reconst_flow.last_generated_outputs['out_gfa']
+            gfa_data = nib.load(gfa_path).get_data()
+            assert_equal(gfa_data.shape, volume.shape[:-1])
+
+            peaks_dir_path =\
+                 reconst_flow.last_generated_outputs['out_peaks_dir']
+            peaks_dir_data = nib.load(peaks_dir_path).get_data()
+            assert_equal(peaks_dir_data.shape[-1], 15)
+            assert_equal(peaks_dir_data.shape[:-1], volume.shape[:-1])
+
+            peaks_idx_path = \
+                reconst_flow.last_generated_outputs['out_peaks_indices']
+            peaks_idx_data = nib.load(peaks_idx_path).get_data()
+            assert_equal(peaks_idx_data.shape[-1], 5)
+            assert_equal(peaks_idx_data.shape[:-1], volume.shape[:-1])
+
+            peaks_vals_path = \
+                reconst_flow.last_generated_outputs['out_peaks_values']
+            peaks_vals_data = nib.load(peaks_vals_path).get_data()
+            assert_equal(peaks_vals_data.shape[-1], 5)
+            assert_equal(peaks_vals_data.shape[:-1], volume.shape[:-1])
+
+            shm_path = reconst_flow.last_generated_outputs['out_shm']
+            shm_data = nib.load(shm_path).get_data()
+            # Test that the number of coefficients is what you would expect
+            # given the order of the sh basis:
+            assert_equal(shm_data.shape[-1],
+                            sph_harm_ind_list(sh_order)[0].shape[0])
```

Comment:
This is where I test that the shape of the output conforms with the input `sh_order` provided.